Sources/_Datadog_Private/ObjcAppLaunchHandler.m

```diff
@@ -56,32 +57,60 @@ @interface AppLaunchHandler : NSObject
 @implementation AppLaunchHandler
 
 + (void)load {
+    pthread_rwlock_init(&rwLock, NULL);
+
     // This is called at the `_Datadog_Private` load time, keep the work minimal
     FrameworkLoadTime = CFAbsoluteTimeGetCurrent();
 
     NSNotificationCenter * __weak center = NSNotificationCenter.defaultCenter;
-    id __block token = [center
-                        addObserverForName:UIApplicationDidBecomeActiveNotification
-                        object:nil
-                        queue:NSOperationQueue.mainQueue
-                        usingBlock:^(NSNotification *_){
-
-        pthread_rwlock_init(&rwLock, NULL);
+    id __weak __block token = [center addObserverForName:UIApplicationDidBecomeActiveNotification
+                                                  object:nil
+                                                   queue:NSOperationQueue.mainQueue
+                                              usingBlock:^(NSNotification *_){
         pthread_rwlock_wrlock(&rwLock);
-        TimeToApplicationDidBecomeActive = ComputeProcessTimeFromStart();
+        UIApplicationDidBecomeActiveNotificationTime = CFAbsoluteTimeGetCurrent();
         pthread_rwlock_unlock(&rwLock);
-
         [center removeObserver:token];
-        token = nil;
     }];
 }
 
 @end
 
+@implementation UIApplication (Tracking)
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        method_exchangeImplementations(
+            class_getInstanceMethod(self, @selector(init)),
+            class_getInstanceMethod(self, @selector(dd_init))
+        );
```

Comment:
AFAIK, this way of swizzling [is dangerous/incorrect](https://newrelic.com/blog/best-practices/right-way-to-swizzle) as it doesn't manage the original  implementation right. In all our swizzlings we [use get/set implementation approach](https://github.com/DataDog/dd-sdk-ios/blob/2c9811e0967cc17da080f88534cf8dfdaaa68590/Sources/Datadog/Core/Swizzling/MethodSwizzler.swift#L58-L74). How are we sure this won't cause any wrong behaviours?

Swizzling the entry point to the program doesn't sound safe IMO. Are we sure this will always perform right, on all platforms and in all setups? E.g. what if someone else already swizzles `UIApplication.init` or the app overrides `UIApplication` class with custom `main` implementation? What about purely SwiftUI apps where `@main` is synthesized in struct?
```swift
@main
struct MyApp: App {
   // ... w/ or w/o init()
}
``` 
I think we need more proofs / tests before considering this safe. Among all our tests we don't really focus on how app is being launched (all are build with classic Storyboard + `AppDelegate` approach).

Reply:
[Very interesting read](https://newrelic.com/blog/best-practices/right-way-to-swizzle), better to sizzle with c function then 👍 

I intuitively don't find it safe either TBO, but I don't want to rely on intuition only and have concrete red flags. This PR is made for discussing those.
> E.g. what if someone else already swizzles UIApplication.init or the app overrides UIApplication class

Theoretically, it shouldn't behave differently than other method swizzling. I will validate that 👍 

> What about purely SwiftUI apps where `@main` is synthesized in struct?

`SwiftUI.App.init()` is called before `UIApplicationMain` from `main()`, so this measure will come after custom code in SwiftUI launch flow! But, from Apple documentation:
> Prewarming executes an app’s launch sequence up until, but not including, when main() calls UIApplicationMain(_:_:_:_:).

The goal here is to get a timestamp at an early stage of `UIApplicationMain()`. Custom code called during `SwiftUI.App.init()` should be called during cold start or during pre-warming. Therefore, they are out of a warm start measure.

> I think we need more proofs / tests before considering this safe.

💯 ! The problem is to get a time as close as possible as when `main()` invoke `UIApplicationMain` and I couldn't get there without swizzling. I'm noting the following validation:
- what if someone else already swizzles `UIApplication.init`
- app overrides UIApplication class ✅ -> SwiftUI uses it's own subclass and `dd_init` is properly called.
- custom implementation of `main()` (objc)